chore: upgrade electron to 32.2.0 (#30394)
cypress-io/cypress: cli/CHANGELOG.md

_Released 12/3/2024 (PENDING)_

**Bugfixes:**

> - The CSS pseudo-class `:dir()` is now supported when testing in Electron. Addresses [#29766](https://github.com/cypress-io/cypress/issues/29766).

Reviewer: @ryanthemanuel This should be moved below breaking changes actually

Author: Fixed: [`56524b7` (#30394)](https://github.com/cypress-io/cypress/pull/30394/commits/56524b7de45f13ec82da008e1814e414250ff684)

**Breaking Changes:**

- Removed support for Node.js 16 and Node.js 21. Addresses [#29930](https://github.com/cypress-io/cypress/issues/29930).
- Prebuilt binaries for Linux are no longer compatible with Linux distributions based on glibc <2.28, for example: Ubuntu 14-18, RHEL 7, CentOS 7, Amazon Linux 2. Addresses [#29601](https://github.com/cypress-io/cypress/issues/29601).
- The undocumented methods `Cypress.backend('firefox:force:gc')` and `Cypress.backend('log:memory:pressure')` were removed. Addresses [#30222](https://github.com/cypress-io/cypress/issues/30222).

> - Upgraded bundled Node.js version from `18.17.0` to `20.18.0`.
> - It is no longer possible to do things like `cy.window().then((win) => win.fetch('<some-url>')` without doing some form of initial navigation via `cy.visit()`. `cy.request` will still work in this scenario.
> 
> **Dependency Updates:**
> 
> - Upgraded `electron` from `27.3.10` to `32.2.0`.
> - Upgraded bundled Chromium version from `118.0.5993.159` to `128.0.6613.178`.

Author: ```suggestion
- Upgraded bundled Node.js version from `18.17.0` to `20.18.0`. Addresses [#29547](https://github.com/cypress-io/cypress/issues/29547).
- It is no longer possible to do things like `cy.window().then((win) => win.fetch('<some-url>')` without doing some form of initial navigation via `cy.visit()`. `cy.request` will still work in this scenario. Addressed in [#29547](https://github.com/cypress-io/cypress/pull/30394).

**Dependency Updates:**

- Upgraded `electron` from `27.3.10` to `32.2.0`. Addresses [#29547](https://github.com/cypress-io/cypress/issues/29547).
- Upgraded bundled Chromium version from `118.0.5993.159` to `128.0.6613.178`. Addresses [#29547](https://github.com/cypress-io/cypress/issues/29547).
```

- Removed support for Node.js 16 and Node.js 21. Addresses [#29930](https://github.com/cypress-io/cypress/issues/29930).
- Prebuilt binaries for Linux are no longer compatible with Linux distributions based on glibc <2.28, for example: Ubuntu 14-18, RHEL 7, CentOS 7, Amazon Linux 2. Addresses [#29601](https://github.com/cypress-io/cypress/issues/29601).
- The undocumented methods `Cypress.backend('firefox:force:gc')` and `Cypress.backend('log:memory:pressure')` were removed. Addresses [#30222](https://github.com/cypress-io/cypress/issues/30222).

> - Upgraded bundled Node.js version from `18.17.0` to `20.18.0`.

Author: Add blurb about having to navigate rather than making fetch or XHR requests in about:blank.

Author: ```suggestion
- Upgraded bundled Node.js version from `18.17.0` to `20.18.0`.
- It is no longer possible to do things like `cy.window().then((win) => win.fetch('<some-url>')` without doing some form of initial navigation. `cy.request` will still work in this scenario.
```

- Removed support for Node.js 16 and Node.js 21. Addresses [#29930](https://github.com/cypress-io/cypress/issues/29930).
- Prebuilt binaries for Linux are no longer compatible with Linux distributions based on glibc <2.28, for example: Ubuntu 14-18, RHEL 7, CentOS 7, Amazon Linux 2. Addresses [#29601](https://github.com/cypress-io/cypress/issues/29601).
- The undocumented methods `Cypress.backend('firefox:force:gc')` and `Cypress.backend('log:memory:pressure')` were removed. Addresses [#30222](https://github.com/cypress-io/cypress/issues/30222).
- Upgraded bundled Node.js version from `18.17.0` to `20.18.0`.

> - It is no longer possible to do things like `cy.window().then((win) => win.fetch('<some-url>')` without doing some form of initial navigation. `cy.request` will still work in this scenario.

Reviewer: ```suggestion
- It is no longer possible to do things like `cy.window().then((win) => win.fetch('<some-url>')` without doing some form of initial navigation via `cy.visit()`. `cy.request` will still work in this scenario.
```

- We removed yielding the second argument of `before:browser:launch` as an array of browser arguments. This behavior has been deprecated since Cypress 4.0.0. Addressed in [#30460](https://github.com/cypress-io/cypress/pull/30460).
- The `cypress open-ct` and `cypress run-ct` CLI commands were removed. Please use `cypress open --component` or `cypress run --component` respectively instead. Addressed in [#30456](https://github.com/cypress-io/cypress/pull/30456)
- The undocumented methods `Cypress.backend('firefox:force:gc')` and `Cypress.backend('log:memory:pressure')` were removed. Addresses [#30222](https://github.com/cypress-io/cypress/issues/30222).
- Upgraded bundled Node.js version from `18.17.0` to `20.18.0`. Addresses [#29547](https://github.com/cypress-io/cypress/issues/29547).

> - It is no longer possible to do things like `cy.window().then((win) => win.fetch('<some-url>')` without doing some form of initial navigation via `cy.visit()`. `cy.request` will still work in this scenario. Addressed in [#29547](https://github.com/cypress-io/cypress/pull/30394).

Reviewer 2: This only applies to electron, correct?

```suggestion
- It is no longer possible to make a `fetch` or `XMLHttpRequest` request from the `about:blank` page in Electron (i.e. `cy.window().then((win) => win.fetch('<some-url>')`). You must use `cy.request` instead or perform some form of initial navigation via `cy.visit()`. Addressed in [#29547](https://github.com/cypress-io/cypress/pull/30394).
```
